code/modules/mafia/controller.dm

```diff
@@ -839,6 +897,27 @@
 	if(GLOB.mafia_signup.len >= MAFIA_MAX_PLAYER_COUNT || custom_setup.len)//enough people to try and make something (or debug mode)
 		basic_setup()
 
+/**
+  * Filters inactive player into a different list until they reconnect, and removes players who are no longer ghosts.
+  *
+  * If a disconnected player gets a non-ghost mob and reconnects, they will be first put back into mafia_signup then filtered by that.
+  */
+/datum/mafia_controller/proc/check_signups()
+	for(var/bad_key in GLOB.mafia_bad_signup)
+		if(GLOB.directory[bad_key] && GLOB.directory[bad_key] == GLOB.mafia_bad_signup[bad_key])
```

Comment:
This is still not correct, you can't compare clients directly. Client's won't always get same ref when reconnecting.

Reply:
THAT'S WHAT I WAS FORGETTING THANK YOU

Reply:
seriously, there was like 3 bad lobby bugs and one just totally blanked on me